src/main/java/com/celements/course/CourseScriptService.java

```diff
@@ -98,49 +100,63 @@ public String passwordHashString(String str) {
   }
 
   public String normalizeEmail(String emailAdr) {
-    return emailAdr.toLowerCase().trim();
+    return courseService.normalizeEmail(emailAdr);
   }
 
-  public boolean validateParticipant(String courseFN, String emailAdr,
-      String activationCode) {
-    DocumentReference partiClassRef = new DocumentReference(getContext().getDatabase(), 
-        "Classes", "CourseParticipantClass");
+  public boolean validateParticipant(String courseFN, String emailAdr, String activationCode) {
+    LOGGER.debug(
+        "validateParticipant with registration doc [{}], email [{}] and activation code [{}]",
+        courseFN, emailAdr, activationCode);
+    DocumentReference partiClassRef = new DocumentReference(getContext().getDatabase(),
+        CourseClasses.COURSE_CLASSES_SPACE, CourseClasses.COURSE_PARTICIPANT_CLASS_DOC);
 
     try {
-      XWikiDocument courseDoc = getContext().getWiki().getDocument(getDocRefForFullName(
-          courseFN), getContext());
-      BaseObject partiObj = courseDoc.getXObject(partiClassRef, "email",
-          normalizeEmail(emailAdr), false);
+      XWikiDocument courseDoc = getContext().getWiki().getDocument(getDocRefForFullName(courseFN),
```

Comment:
please change to use modelAccess. getDocument with changing the document is potentially spoiling the document cache.